frmts/tiledb/tiledbsparse.cpp

```diff
@@ -0,0 +1,4934 @@
+                                    std::max<size_t>(1,
+                                                     static_cast<size_t>(
+                                                         nEstBytes / nEstRows) *
+                                                         4 / 3);
+                                CPLDebug("TILEDB", "Average %s size: %u bytes",
+                                         pszFieldName,
+                                         static_cast<unsigned>(
+                                             m_oMapEstimatedSizePerRow
+                                                 [pszFieldName]));
+                            }
+                            break;
+                        }
+
+                        default:
+                            break;
+                    }
+                }
+
+                m_nEstimatedWkbSizePerRow = 9;  // Size of 2D point WKB
+                if (pszGeomColName)
+                {
+                    const auto estimation =
+                        m_query->est_result_size_var(pszGeomColName);
+                    const uint64_t nEstRows = estimation[0] / sizeof(uint64_t);
+                    const uint64_t nEstBytes = estimation[1];
+                    if (nEstRows)
+                    {
+                        m_nEstimatedWkbSizePerRow = std::max(
+                            m_nEstimatedWkbSizePerRow,
+                            static_cast<size_t>(nEstBytes / nEstRows) * 4 / 3);
+                        CPLDebug(
+                            "TILEDB", "Average WKB size: %u bytes",
+                            static_cast<unsigned>(m_nEstimatedWkbSizePerRow));
+                    }
+                }
+            }
+
+            if (m_poFilterGeom && queryCondition == nullptr)
+            {
+                tiledb::Subarray subarray(*m_ctx, *m_array);
+
+                const double dfMinX =
+                    std::max(m_dfXStart, m_sFilterEnvelope.MinX - m_dfPadX);
+                const double dfMaxX =
+                    std::min(m_dfXEnd, m_sFilterEnvelope.MaxX + m_dfPadX);
+                const double dfMinY =
+                    std::max(m_dfYStart, m_sFilterEnvelope.MinY - m_dfPadY);
+                const double dfMaxY =
+                    std::min(m_dfYEnd, m_sFilterEnvelope.MaxY + m_dfPadY);
+
+                if (dfMaxX < dfMinX || dfMaxY < dfMinY)
+                {
+                    m_bQueryComplete = true;
+                    return false;
+                }
+
+                subarray.add_range(m_osXDim, dfMinX, dfMaxX);
+                subarray.add_range(m_osYDim, dfMinY, dfMaxY);
+                m_query->set_subarray(subarray);
+            }
+        }
+
+        SetReadBuffers();
+
+        // Create a loop
+        tiledb::Query::Status status;
+        uint64_t nRowCount = 0;
+        while (true)
+        {
+            // Submit query and get status
+            m_query->submit();
```

Comment:
Missing stats as per the dense implementation https://github.com/rouault/gdal/blob/tiledb_geometries_ero/frmts/tiledb/tiledbdense.cpp#L1938 it will need to wrap `m_query→submit()` and be a configurable creation/open option.